quai/api_backend.go

```diff
@@ -574,9 +574,27 @@ func (b *QuaiAPIBackend) SubscribeExpansionEvent(ch chan<- core.ExpansionEvent)
 	return b.quai.core.SubscribeExpansionEvent(ch)
 }
 
-// ///////////////////////////
-// /////// P2P ///////////////
-// ///////////////////////////
-func (b *QuaiAPIBackend) BroadcastBlock(block *types.WorkObject, location common.Location) error {
-	return b.quai.p2p.Broadcast(location, block)
+//////////////////////////////
+////////// P2P ///////////////
+//////////////////////////////
+func (b *QuaiAPIBackend) BroadcastWorkObject(wo *types.WorkObject, location common.Location) error {
+	errChan := make(chan error, 2)
+
+	go func() {
+		err := b.quai.p2p.Broadcast(location, wo.ConvertToBlockView())
+		errChan <- err
+	}()
+
+	go func() {
+		err := b.quai.p2p.Broadcast(location, wo.ConvertToHeaderView())
+		errChan <- err
+	}()
+
+	// Wait for errors from both goroutines
+	for i := 0; i < 2; i++ {
```

Comment:
where is the wait logic in this? I see that if one of them throws the error we just return, i dont see any waitgroup logic which will help the wait

Reply:
True the comment says we wait but we don't. I did it that way because we don't really care what the other one does if we get an error in one.